pkg/resources/configcheck/configcheck.go

```diff
@@ -0,0 +1,101 @@
+// Copyright © 2023 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configcheck
+
+import (
+	"context"
+	"emperror.dev/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const hashLabel = "logging.banzaicloud.io/config-hash"
+
+func WithHashLabel(accessor v1.Object, hash string) {
+	l := accessor.GetLabels()
+	l[hashLabel] = hash
```

Comment:
`l` might be `nil` here
```suggestion
	if l == nil {
		l = map[string]string{}
	}
	l[hashLabel] = hash
```